src/mappings/erc20Templates.ts

```diff
@@ -204,6 +198,35 @@ export function handleNewPaymentCollector(event: NewPaymentCollector): void {
   token.save()
 }
 
+export function handleProviderFee(event: ProviderFee): void {
+  const providerFee: string = `{"providerFeeAddress": "${event.params.providerFeeAddress.toHex()}", "providerFeeToken": "${event.params.providerFeeToken.toHex()}", "providerFeeAmount": "${
+    event.params.providerFeeAmount
+  }", "providerData": "${event.params.providerData.toHexString()}", "v": "${
+    event.params.v
+  }", "r": "${event.params.r.toHexString()}", "s": "${event.params.s.toHexString()}", "validUntil": "${
+    event.params.validUntil
+  }"}`
+
+  const orderId = getOrderId(
+    event.transaction.hash.toHex(),
+    event.address.toHex(),
+    event.transaction.from.toHex()
+  )
+  const order = Order.load(orderId)
+  let orderReuse = OrderReuse.load(event.transaction.hash.toHex())
+  if (order) {
+    order.providerFee = providerFee
+    order.save()
+  } else if (orderReuse) {
```

Comment:
maybe move the orderReuse load to before this line, so would only perform data load if order not found
so probably like below:
let orderReuse = OrderReuse.load(event.transaction.hash.toHex())
if (orderReuse) {
orderReuse.providerFee = providerFee
    orderReuse.save()
  }

then there is no more else condition